### ISSUE TITLE
ImportError and execution issue with DuckDB on Python 3.7.9 in aarch64 CentOS and Python 3.6.9 in aarch64 Ubuntu environment

### ISSUE THREAD

**Issue by lq1909017169** [duckdb/duckdb#9285]
### What happens?

I encountered some issues while using DuckDB and would appreciate your help and guidance. Here is a description of the problems I experienced:

1，I installed Python 3.7.9 in aarch64 CentOS environment and attempted to import DuckDB module. However, an error occurred as follows:

ImportError: Cut of Memory Error: Failed to allocate block of 2048 bytes
This error seems to be related to memory allocation, which prevented me from importing DuckDB successfully.

2，In another environment running aarch64 Ubuntu, I installed Python 3.6.9 and DuckDB 0.8.1. Although I was able to import DuckDB successfully, when I tried to execute a simple query duckdb.sql('SELECT 42').show(), there was no output. Instead, the operation seemed to make the process unresponsive, and I had to manually terminate it.

I have tried updating pip and made sure that the necessary build dependencies were installed on my system. However, these issues persist.

### To Reproduce

1，[root@ecs-3cCa -]# python3
Python 3.7.9 (default, 0ct 9 2023,22:59:00)
[GOC 8.3.1 20190507 (Ped Hat 8,3,1-4)] on linux
lype "help","copyright","credits" or "icense" for more information.
>>> import duckdb
Traceback (most recent call last):
File " stdin>", line 1, in <module>
ImportError: Cut of Memory Error: Failed to allocate block of 2048 bytes
2，root@ecs-1181:~# python3
Python 3.6.9 (default, Mar 10 2023, 16:46:00) 
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import duckdb
>>> duckdb.sql('SELECT 42').show()

### OS:

aarch64

### DuckDB Version:

0.8.1

### DuckDB Client:

python

### Full Name:

hiten

### Affiliation:

Ordinary People

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by Mause** [comment]
Hey there, what's the page size configured as on your system?

**Comment by Mause** [comment]
I ask as this sounds like a duplicate of https://github.com/duckdb/duckdb/issues/8095

**Comment by lq1909017169** [comment]
> Hey there, what's the page size configured as on your system?

[root@ecs-1198 ~]# getconf PAGESIZE
65536

**Comment by szarnyasg** [comment]
Hi @lq1909017169 unfortunately Python versions <3.8 are not supported, see the [Python documentation page](https://duckdb.org/docs/api/python/overview):
> You must be using Python 3.8 or newer.

Did the error occur with Python 3.8+?

**Comment by szarnyasg** [comment]
Apologies - @Mause pointed out that Python 3.7 should also work. So disregard my previous comment.

**Comment by szarnyasg** [comment]
I don't have access to a CentOS machine but I'm unable to reproduce it on CentOS running on a MacBook Pro with and M2Pro CPU using a CentOS 7 Docker image.

**Comment by szarnyasg** [comment]
I managed to reproduce it. For me, even `import duckdb` crashes:

To reproduce:

1. Run a CentOS 7 aarch64 instance with AMI `ami-0144a5a84f5699847` on a t4g.2xlarge instance.
2. Install Python 3.7 as described in the second half of https://vinodpandey.com/how-to-install-python3-on-centos-7/#contents:step-2-downloading-python-36
3. Run:

```bash
python3.7 -c "import duckdb"
```
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/centos/.local/lib/python3.7/site-packages/duckdb/__init__.py", line 4, in <module>
    import duckdb.functional as functional
  File "/home/centos/.local/lib/python3.7/site-packages/duckdb/functional/__init__.py", line 1, in <module>
    from duckdb.duckdb.functional import (
ImportError: Out of Memory Error: Failed to allocate block of 2048 bytes
```

**Comment by Mause** [comment]
Duplicate of https://github.com/duckdb/duckdb/issues/8095

**Comment by szarnyasg** [comment]
@lq1909017169 I described the steps to manually compile DuckDB for CentOS7 / aarch64 / Python 3.7 at https://github.com/duckdb/duckdb/issues/8095#issuecomment-1805656569.
Note that you need Python 3.7, 3.6 is not sufficient.

**Comment by szarnyasg** [comment]
This was resolved by #9642
